.github/workflows/ci.yml

```diff
@@ -18,24 +18,35 @@ jobs:
         uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
-      - name: Cache node modules
-        uses: actions/cache@v3
-        env:
-          cache-name: cache-node-modules
-        with:
-          path: node_modules
-          key: ${{ runner.os }}-node${{ runner.node-version }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
       - run: npm ci
       - run: npm run build
       - run: npm test
+      # Upload coverage for sonarcube
+      - uses: actions/upload-artifact@v3
```

Comment:
I think this is still conditional, as otherwise we upload N artifacts per run, N being systems in the above matrix